submit-api/src/submit_api/models/queries/account_project.py

```diff
@@ -184,8 +192,39 @@ def _filter_by_search_text(cls, query, search_text):
 
     @classmethod
     def _filter_by_submission_status(cls, query, statuses):
-        """Filter by submission status using overlap."""
-        return query.filter(Package.status.op("@>")([status.value for status in statuses]))
+        """Filter by submission status, with special handling for revision required."""
+        revision_required_value = PackageStatus.REVISION_REQUIRED.value
+
+        # Separate normal statuses and check if revision_required is included
+        normal_statuses = [
+            status.value for status in statuses if status.value != revision_required_value]
+        include_revision_required = any(
+            status.value == revision_required_value for status in statuses)
+
+        if normal_statuses:
+            query = query.filter(Package.status.op("@>")(normal_statuses))
+
+        if include_revision_required:
+            query = cls._revision_required_filter(query)
+
+        return query
+
+    @classmethod
+    def _revision_required_filter(cls, query):
+        """Joins UpdateRequest and filters for packages requiring revision."""
+        return query.filter(
+            ~Package.status.op("@>")([
+                PackageStatus.COMPLETED.value,
+                PackageStatus.PARTIALLY_COMPLETED.value,
+            ])
+        ).join(
+            UpdateRequest,
+            and_(
```

Comment:
I don't think this and should be part of the join operation. It probably should be in a filter after the join